contracts/vesting/src/vesting.rs

```diff
@@ -0,0 +1,559 @@
+use crate::contract::{NUM_VESTING_PERIODS, VESTING_PERIOD};
+use crate::errors::ContractError;
+use crate::storage::{
+    get_account_balance, get_account_delegations, set_account, set_account_balance,
+    set_account_delegations,
+};
+use config::defaults::{DEFAULT_MIXNET_CONTRACT_ADDRESS, DENOM};
+use cosmwasm_std::{Addr, Coin, DepsMut, Env, QuerierWrapper, Storage, Timestamp, Uint128};
+use mixnet_contract::ExecuteMsg as MixnetExecuteMsg;
+use mixnet_contract::IdentityKey;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+pub trait VestingAccount {
+    // locked_coins returns the set of coins that are not spendable (can still be delegated tough) (i.e. locked),
+    // defined as the vesting coins that are not delegated.
+    //
+    // To get spendable coins of a vesting account, first the total balance must
+    // be retrieved and the locked tokens can be subtracted from the total balance.
+    // Note, the spendable balance can be negative.
```

Comment:
That is from cosmos interface spec, my interpretation is that since delegations subtract from the balance but still count in the locked tokens that can go <= 0. I plan to double check that in their Go code